### PR TITLE
Improving quality of HPX build

### PR DESCRIPTION
**Pull request by sithhell**
- Adding jemalloc as allocator
 - Forwarding the proper build type
 - Setting build type to release in QUICKSTART script

### CODE REVIEW

File: cmake/allscale_runtime.cmake [hunk 1]
```diff
@@ -15,9 +20,11 @@ ExternalProject_Add(
 		-DHPX_WITH_THREAD_IDLE_RATES=On
 		-DBOOST_ROOT=${Boost_INCLUDE_DIRS}/..
 		-DHWLOC_ROOT=${Hwloc_INCLUDE_DIRS}/..
+        -DJEMALLOC_FOUND=True
+        -DJEMALLOC_LIBRARIES=${JEMALLOC_}
```
Comment (W4RH4WK) [inline cmake/allscale_runtime.cmake]: Shouldn't this be `JEMALLOC_LIBRARY` ?

Comment (sithhell) [reply cmake/allscale_runtime.cmake]: Correct. This is now fixed.

**Comment by W4RH4WK** [comment]
I have done some slight modifications, please have a look and if they are okay with you.

**Comment by sithhell** [comment]
Looks good, thanks!

**Comment by W4RH4WK** [comment]
It seems like this change causes the runtime library name to change. Before this, it was `liballscale.so`, now it is `libhpx_allscale.so`. Yet the runtime submodule did not change.

Do you have an idea what could have changed this? Maybe I am mistaken and the error does not come from this commit.

**Comment by sithhell** [comment]
No, I have no idea where this comes from. The name was alwaus `libhpx_allscale.so`.

**Comment by philippgs** [comment]
Nope, it wasn't. I built the project today in the morning from scratch using commit `59cadb3255d00a92b57853f79107246139f693f5` of the allscale-compiler project, and this is the path and name of the lib in my build directory:
`allscale_runtime-prefix/src/allscale_runtime-build/src/liballscale.so`

**Comment by sithhell** [comment]
I have a few more changes incoming regarding the build settings. The build type mismatches are really severe.

**Comment by sithhell** [comment]
>  Nope, it wasn't. I built the project today in the morning from scratch using commit 59cadb3255d00a92b57853f79107246139f693f5 of the allscale-compiler project, and this is the path and name of the lib in my build directory:
allscale_runtime-prefix/src/allscale_runtime-build/src/liballscale.so

*shrug* it has always been libhpx_allscale.so when the runtime was built standalone. I have no idea what was messing up the names previously.

**Comment by W4RH4WK** [comment]
I think I found the culprit. With these changes we now set `CMAKE_BUILD_TYPE` for the AllScale Runtime , which is set to `Release` by default in our `cmake/build_settings.cmake`.

@sithhell You are correct that the library should indeed be named `libhpx_allscale.so`, yet for us it was always `liballscale.so` since we have never set the build type before.

More interestingly, I found that the bottom layer of this error seems to be in HPX. Their CMake infrastructure basically relies on having a build type set -- yet the logic for setting the default build type is flawed.

See STEllAR-GROUP/hpx#3033.

**Comment by W4RH4WK** [comment]
Posted a detailed description to STEllAR-GROUP/hpx#3033, which also mentions this issue.

**Comment by sithhell** [comment]
I am completely clueless as to why a unset build type would lead to an omission of the `hpx_` prefix in the library name... Let's fix this now.

For the time being, I'd also suggest to fix the build type for HPX and AllScale Runtime to release, as a default, this avoids various problems due to mismatch of debug vs. release build flags and should get us going for a short term solution. Please see #4 for further discussion on how to improve the situation overall.

**Comment by W4RH4WK** [comment]
This was merged in a3b9778, yet I just discovered that we need to forward different Jemalloc flags. I am testing this locally right now, should be live soon.

**Comment by sithhell** [comment]
Ok, merging this just yet was not a good idea ;)
You will get allscalecc failures if you set `CMAKE_BUILD_TYPE=Debug` since the HPX build then generates a `libhpxd.so` library instead of `libhpx.so`. This was the main motivation to suggest to fix the HPX and AllScale Runtime build to be Release, to match the compiler flags set in allscalecc and the integration tests. Everything else would generate more subtile problems at this point in time.

**Comment by W4RH4WK** [comment]
The correct Jemalloc flags are now passed with f02da19.

I set the build type for HPX and the AllScale Runtime to `Release` for now, we'll change this together with #4.

**Comment by sithhell** [comment]
This has been resolved on master.

**Pull request closed without merging**